packages/expo-modules-core/ios/Tests/PropertyComponentSpec.swift

```diff
@@ -93,6 +93,44 @@ class PropertyComponentSpec: ExpoSpec {
         expect(value?.kind) == .number
         expect(value?.getInt()) == newValue
       }
+
+      // Tests for accessing shared object properties through KeyPath and ReferenceWritableKeyPath
+      describe("key path") {
+        it("gets immutable property") {
+          let value = try runtime?.eval([
+            "object = new expo.modules.PropertyTest.TestClass()",
+            "object.immutableKeyPathProperty"
+          ])
+
+          expect(value?.kind) == .number
+          expect(value?.getInt()) == TestClass.constantValue
+        }
+
+        it("cannot set immutable property") {
+          let newValue = Int.random(in: 100..<200)
+          let value = try runtime?.eval([
+            "object = new expo.modules.PropertyTest.TestClass()",
+            "object.immutableKeyPathProperty = \(newValue)",
+            "object.immutableKeyPathProperty"
+          ])
+
+          // Returned value didn't change, it doesn't equal to `newValue`
```

Comment:
nit: would be even better if updating immutable property could have a compiler-time or build-time error. not sure whether it's feasible or not.

Reply:
No, it doesn't seem possible, it's just a JS 🤷‍♂️ We can throw an error at runtime, but the current implementation is the same as in JS. Even `Object.defineProperty(obj, "key", { writable: false })` still lets you do the assignment operation, but it's just a no-op and will return `undefined` when you read that property.

Also, I think it would be nice to add some attributes to the `Property`, such as `writable`, `enumerable` and `readonly`. The last one is not a part of the descriptor but it could change the behavior so that we throw an error when you try setting that property.

Reply:
the `Object.defineProperty(obj, "key", { writable: false })` with no-op setter makes sense to me. thanks!